bin/test-workflow/8_app_verify_authentication.sh

```diff
@@ -0,0 +1,183 @@
+#!/usr/bin/env bash
+set -euo pipefail
+
+. utils.sh
+
+init_bash_lib
+
+RETRIES=150
+# Seconds
+RETRY_WAIT=2
+
+# Dump some kubernetes resources and Conjur authentication policy if this
+# script exits prematurely
+DETAILED_DUMP_ON_EXIT=true
+
+function finish {
+  readonly PIDS=(
+    "SIDECAR_PORT_FORWARD_PID"
+    "INIT_PORT_FORWARD_PID"
+    "INIT_WITH_HOST_OUTSIDE_APPS_PORT_FORWARD_PID"
+    "SECRETLESS_PORT_FORWARD_PID"
+  )
+
+  if [[ "$DETAILED_DUMP_ON_EXIT" == "true" ]]; then
+    dump_kubernetes_resources
+    dump_authentication_policy
+  fi
+
+  set +u
+
+  echo -e "\n\nStopping all port-forwarding"
+  for pid in "${PIDS[@]}"; do
+    if [ -n "${!pid}" ]; then
+      # Kill process, and swallow any errors
+      kill "${!pid}" > /dev/null 2>&1
+    fi
+  done
+}
+trap finish EXIT
+
+announce "Validating that the deployments are functioning as expected."
+
+set_namespace "$TEST_APP_NAMESPACE_NAME"
+
+deploy_test_curl() {
+  $cli delete --ignore-not-found pod/test-curl
+  $cli create -f ./$PLATFORM/test-curl.yml
+}
+
+check_test_curl() {
+  pods_ready "test-curl"
+}
+
+pod_curl() {
+  kubectl exec test-curl -- curl "$@"
+}
+
+if [[ "$TEST_APP_LOADBALANCER_SVCS" == "false" ]]; then
+  echo "Deploying a test curl pod"
+  deploy_test_curl
+  echo "Waiting for test curl pod to become available"
+  bl_retry_constant "${RETRIES}" "${RETRY_WAIT}"  check_test_curl
+fi
+  
+echo "Waiting for pods to become available"
+
+check_pods(){
+  # pods_ready "test-app-summon-init" &&
+  # pods_ready "test-app-with-host-outside-apps-branch-summon-init" &&
+  pods_ready "test-app-summon-sidecar" # &&
+  # pods_ready "test-app-secretless"
+}
+bl_retry_constant "${RETRIES}" "${RETRY_WAIT}"  check_pods
+
+if [[ "$PLATFORM" == "openshift" ]]; then
+  echo "Waiting for deployments to become available"
+
+  check_deployment_status(){
```

Comment:
I think we can also delete the `check_deployment_status()` function since we're using `--wait` on all of the Helm installs?

Reply:
Done